retriever/lib/datapackage.py

```diff
@@ -247,9 +247,9 @@ def create_json():
     file_name = contents['name'] + ".json"
     file_name = file_name.replace('-', '_')
     with open(os.path.join(HOME_DIR, 'scripts', file_name), 'w') as output_file:
-        json_str = str(json.dump(contents, output_file, sort_keys=True, indent=4,
-                              separators=(',', ': ')))
-        output_file.write(json_str + '\n')
+        json.dump(contents, output_file, sort_keys=True, indent=4,
+                  separators=(',', ': '))
+        output_file.write('\n')
         print("\nScript written to " + file_name)
```

Comment:
looks fine. 
Could you change `for key, val in obj.items():` to `for key, val in obj.copy().items():`
https://github.com/weecology/retriever/pull/1306/files#diff-7cb8d9949f19aa110e816b90e44ffb2dR261

Reply:
Done.